packages/rocketchat-lib/client/lib/openRoom.js

```diff
@@ -65,6 +65,13 @@ function openRoom(type, name) {
 			Session.set('openedRoom', room._id);
 			RocketChat.openedRoom = room._id;
 
+			const room_type = RocketChat.roomTypes.roomTypes[type].route.name;
+			const room_path = `/${ room_type }/${ name }`;
+
+			if (Meteor.user().settings && Meteor.user().settings.preferences.saveLastVisitedRoom) {
+				Meteor.call('setLastVisitedRoom', Meteor.userId(), room_path);
```

Comment:
Pass the path is not the best option, room id will be better cuz room name and type can change.